roles/sap_install_media_detect/tasks/prepare/create_file_list.yml

```diff
@@ -41,13 +54,20 @@
   ansible.builtin.debug:
     var: __sap_install_media_detect_fact_files_without_extension_file_results
 
+- name: SAP Install Media Detect - Prepare - Set fact with file names without extension and of type 'Zip archive data'
```

Comment:
From my point of view repetition is not needed here. One can process all the file types in one nested loop.

One of the ways to do so might be:

```ansible
- name: SAP Install Media Detect - Prepare - Set fact with file names without extension for all supported file types 
  ansible.builtin.set_fact:
   __files_without_extension:
     extension: "{{ __files_without_extension[extension] + [filename] }}"
  when: filename.file_type is search filetype_mapping[extension]
  loop: "{{ __sap_install_media_detect_fact_files_without_extension_file_results | product(filetype_mapping.keys() )| list }}" 
  vars:  
     filetype_mapping: # this variable should be in role vars to make it configurable, here just for reference
       zip: 'Zip archive data'
       exe: 'RAR self-extracting archive'
       rar: 'RAR archive data'
     filename: "{{ item.0 }}"
     extension: "{{ item.1 }}"
       

 # __files_without_extension - dictionary of lists
# __files_without_extension['rar'] - list of all rar files
# __files_without_extension['exe'] - list of all exe files
# this should also simplify later loop for any number of extensions, for instance if sar will be added. 
# __files_without_extension should be initialised as 
# __files_without_extension:
#     zip: []
#     exe: []
#.    rar: []
```

Code should "perform" the same as three separate checks, because number of actual checks is still the same. Benefit it is one task. 

To be honest I did not try code above, but it should work from my point of view. If not, it is simpler to remove nested loop and include task file with set_facts in a loop for filetype_mapping

Reply:
@kksat Many thanks for these suggestions, but I do not want to spend more time right now on trying to fully understand and re-test the code you provided. I think I will do this - if time permits - if I finished the basic rework of the role.